tools/hidl/hidlcomp.cpp

```diff
@@ -3479,8 +3479,22 @@ void EspMessageInfo::write_esp()
     indentOuts(-1,"}\n");
         
     indentOuts("if (flags & 0x01) {\n");
+    bool isExtComplexType = true; //a complex type with no children, no attribute and no parent
+    if (isExtSimpleType || getParentName() || hasNonAttributeChild() || (espm_type_==espm_response && getMetaInt("exceptions_inline", 0)))
```

Comment:
Not sure why it would be done, but just wondering if getParentName() might be set, but the parent is also an empty complex type?

That is empty + emptyParent = empty?

Or maybe the "all" shows up in the parent then?